File: php/ExampleAuthenticate.php
<?php

require_once('Tinkerforge/IPConnection.php');

use Tinkerforge\IPConnection;
use Tinkerforge\TinkerforgeException;

const HOST = 'localhost';
const PORT = 4223;
const SECRET = 'My Authentication Secret!';

// Authenticate each time the connection got (re-)established
function cb_connected($connectReason, $userData)
{
	$ipcon = $userData;

	switch ($connectReason) {
	case IPConnection::CONNECT_REASON_REQUEST:
		echo "Connected by request\n";
		break;
	}

	// Authenticate first...
	try {
		$ipcon->authenticate(SECRET);
		echo "Authentication succeeded\n";
	} catch (TinkerforgeException $e) {
		echo "Could not authenticate: $e\n";
		return;
	}

	// ...then trigger enumerate
	$ipcon->enumerate();
}

// Print incoming enumeration
function cb_enumerate($uid, $connectedUid, $position,
                      $hardwareVersion, $firmwareVersion,
                      $deviceIdentifier, $enumerationType)
{
	echo "UID: $uid, Enumeration Type: $enumerationType\n";
}

// Create IP connection and connect to brickd
$ipcon = new IPConnection();

// Register connected callback to "cb_connected"
$ipcon->registerCallback(IPConnection::CALLBACK_CONNECTED, 'cb_connected', $ipcon);

// Register enumerate callback to "cb_enumerate"
$ipcon->registerCallback(IPConnection::CALLBACK_ENUMERATE, 'cb_enumerate');

$ipcon->connect(HOST, PORT);

echo "Press ctrl+c to exit\n";
$ipcon->dispatchCallbacks(-1); // Dispatch callbacks forever

?>

File: csharp/ExampleAuthenticate.cs
using Tinkerforge;

class ExampleAuthenticate
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string SECRET = "My Authentication Secret!";

	// Authenticate each time the connection got (re-)established
	static void ConnectedCB(IPConnection sender, short connectReason)
	{
		switch(connectReason)
		{
			case IPConnection.CONNECT_REASON_REQUEST:
				System.Console.WriteLine("Connected by request");
				break;

			case IPConnection.CONNECT_REASON_AUTO_RECONNECT:
				System.Console.WriteLine("Auto-Reconnected");
				break;
		}

		// Authenticate first...
		try
		{
			sender.Authenticate(SECRET);
			System.Console.WriteLine("Authentication succeeded");
		}
		catch(TinkerforgeException)
		{
			System.Console.WriteLine("Could not authenticate");
			return;
		}

		// ...then trigger enumerate
		sender.Enumerate();
	}

	// Print incoming enumeration
	static void EnumerateCB(IPConnection sender,
	                        string uid, string connectedUid, char position,
	                        short[] hardwareVersion, short[] firmwareVersion,
	                        int deviceIdentifier, short enumerationType)
	{
		System.Console.WriteLine("UID: " + uid + ", Enumeration Type: " + enumerationType);
	}

	static void Main()
	{
		// Create IP Connection
		IPConnection ipcon = new IPConnection();

		// Register Connected Callback
		ipcon.Connected += ConnectedCB;

		// Register Enumerate Callback
		ipcon.EnumerateCallback += EnumerateCB;

		// Connect to brickd
		ipcon.Connect(HOST, PORT);

		System.Console.WriteLine("Press key to exit");
		System.Console.ReadKey();
		ipcon.Disconnect();
	}
}
